script.js

```diff
@@ -1,12 +1,602 @@
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  crimeShows.innerHTML = crimeShowsHtml;
+  let actionShowsHtml = allActionShows.slice([0],[10]).map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  actionShows.innerHTML = actionShowsHtml;
+  let scifiShowsHtml = allScifiShows.slice([0],[10]).map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  scifiShows.innerHTML = scifiShowsHtml;
+
+  selectShow(allShows);
+
+  const logo = document.getElementById('logo');
+
+  logo.addEventListener('click', () => {
+    changeFormSection();
+    showContainer.className = "row show-content"
+    onLoadScreen(allShows);
+  });
+
+  loadShow();
+}
+
+function loadShow() {
+  const showName = document.getElementsByClassName('show-name');
+
+  for (let i = 0; i < showName.length; i++) {
+    const clickedShow = showName[i];
+    clickedShow.addEventListener('click', (e) => {
+      const showClickedOn= e.target;
+      const showId = showClickedOn.id;
+    freeSearchCont.style.display = 'none';
+    seasonInputContainer.style.display = 'flex';
+    showSetup(showId);
+    });
+  }
+};
+
+// setup fetch to get episodes
+function showSetup(id) {
+  allShowsContainer.style.display = 'none';
+  showContainer.style.display = 'block';
+  let seasonEpisodes = document.getElementById('seasonEpisodes');
+  seasonEpisodes.className = "card-group row row-cols-1 row-cols-md-2 row-cols-lg-3 g-4";
+  let allShows = getAllShows();
+  let showUrl = allShowInfoUrl.replace('SHOW_ID', id);
+
+  fetch(showUrl)
+    .then((res) => {
+      if (!res.ok){
+        throw error (`Error Status: ` + res.status)
+      }
+      return res.json();
+    })
+    .then((data) => {
+      let showInfo = data;
+      let episodes = data['_embedded'].episodes;
+      let cast = data['_embedded'].cast;
+
```

Comment:
I would like to discuss what this code `let episodes = data['_embedded'].episodes;` exactly does.

Reply:
gets the list of episode from the shows embedded API call. here's a link https://www.tvmaze.com/api#embedding